src/main/java/org/ow2/proactive/catalog/rest/controller/BucketGrantController.java

```diff
@@ -107,14 +127,17 @@ public BucketGrantMetadata updateBucketGrantForAUser(
                             @ApiResponse(code = 403, message = "Permission denied"), })
     @RequestMapping(value = "/{bucketName}/grant/group", method = PUT)
     @ResponseStatus(HttpStatus.OK)
+    @Transactional
     public BucketGrantMetadata updateBucketGrantForAGroup(
             @ApiParam(value = "The session id used to access ProActive REST server.", required = true) @RequestHeader(value = "sessionID", required = true) String sessionId,
             @ApiParam(value = "The name of the bucket where the catalog objects are stored.", required = true) @PathVariable String bucketName,
             @ApiParam(value = "The name of the group of users that are benefiting from the access grant.", required = true, defaultValue = "") @RequestParam(value = "userGroup", required = true, defaultValue = "") String userGroup,
-            @ApiParam(value = "The new type of the access grant. It can be either read, write or admin.", required = true, defaultValue = "") @RequestParam(value = "accessType", required = true, defaultValue = "") String accessType)
+            @ApiParam(value = "The new type of the access grant. It can be either noAccess, read, write or admin.", required = true, defaultValue = "") @RequestParam(value = "accessType", required = true, defaultValue = "") String accessType,
+            @ApiParam(value = "The new priority of the access grant. It can be a value from 1 to 10.", required = true) @RequestParam(value = "priority", required = true) int priority)
```

Comment:
The priority of the new access grant. It can be a value from 1 (lowest) to 10 (highest), with 5 as default.
Priorities are used to compute the final acces rights of a user belonging to multiple groups. Group grants with the same priority will resolve with the default accessType order (admin > write > read > noaccess).
Finally, please note that a user grant has always more priority than a group grant.